website/docs/integrations/home_assistant.md

```diff
@@ -648,84 +668,117 @@ Don't forget to replace `<teslamate url>` and `<your tesla model>` with correct
- - platform: template
-   sensors:
+- platform: template
+  sensors:
     tesla_est_battery_range_mi:
       friendly_name: Estimated Range (mi)
       unit_of_measurement: mi
       icon_template: mdi:gauge
+      availability_template: "{{ has_value('sensor.tesla_est_battery_range_km') }}"
       value_template: >
-       {{ (states('sensor.tesla_est_battery_range_km') | float / 1.609344) | round(2) }}
+        {% if has_value('sensor.tesla_est_battery_range_km') %}
+          {{ (states('sensor.tesla_est_battery_range_km') | float / 1.609344) | round(2) }}
+        {% endif %}
 
     tesla_rated_battery_range_mi:
       friendly_name: Rated Range (mi)
       unit_of_measurement: mi
       icon_template: mdi:gauge
+      availability_template: "{{ has_value('sensor.tesla_rated_battery_range_km') }}"
       value_template: >
-       {{ (states('sensor.tesla_rated_battery_range_km') | float / 1.609344) | round(2) }}
+        {% if has_value('sensor.tesla_rated_battery_range_km') %}
+          {{ (states('sensor.tesla_rated_battery_range_km') | float / 1.609344) | round(2) }}
+        {% endif %}
 
     tesla_ideal_battery_range_mi:
       friendly_name: Ideal Range (mi)
       unit_of_measurement: mi
       icon_template: mdi:gauge
+      availability_template: "{{ has_value('sensor.tesla_ideal_battery_range_km') }}"
       value_template: >
-       {{ (states('sensor.tesla_ideal_battery_range_km') | float / 1.609344) | round(2) }}
+        {% if has_value('sensor.tesla_ideal_battery_range_km') %}
+          {{ (states('sensor.tesla_ideal_battery_range_km') | float / 1.609344) | round(2) }}
+        {% endif %}
 
     tesla_odometer_mi:
       friendly_name: Odometer (mi)
       unit_of_measurement: mi
       icon_template: mdi:counter
+      availability_template: "{{ has_value('sensor.tesla_odometer') }}"
       value_template: >
-       {{ (states('sensor.tesla_odometer') | float / 1.609344) | round(2) }}
+        {% if has_value('sensor.tesla_odometer') %}
+          {{ (states('sensor.tesla_odometer') | float / 1.609344) | round(2) }}
+        {% endif %}
 
     tesla_speed_mph:
       friendly_name: Speed (MPH)
       unit_of_measurement: mph
       icon_template: mdi:speedometer
+      availability_template: "{{ has_value('sensor.tesla_speed') }}"
       value_template: >
-       {{ (states('sensor.tesla_speed') | float / 1.609344) | round(2) }}
+        {% if has_value('sensor.tesla_speed') %}
+          {{ (states('sensor.tesla_speed') | float / 1.609344) | round(2) }}
+        {% endif %}
 
     tesla_elevation_ft:
       friendly_name: Elevation (ft)
       unit_of_measurement: ft
       icon_template: mdi:image-filter-hdr
+      availability_template: "{{ has_value('sensor.tesla_elevation') }}"
       value_template: >
-       {{ (states('sensor.tesla_elevation') | float * 3.2808 ) | round(2) }}
+        {% if has_value('sensor.tesla_elevation') %}
+          {{ (states('sensor.tesla_elevation') | float / 3.2808) | round(2) }}
```

Comment:
Did you mean to change the * operation to divide here?

Reply:
Sorry mistake from copy paste.